File: javascript/medium/groupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const map = new Map(), res = new Array();
    for(let str of strs) {
        let key = str.split('').sort().join('');
        if(!map.has(key)) {
            map.set(key, new Array());
        }
        map.get(key).push(str);
    }
    map.forEach(function(v, k, map){
        res.push(v);
    });
    return res;
};

File: ruby/medium/group_anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    map = {}
    res = []
    strs.each do |str|
        key = str.chars.sort.join
        if !map.has_key?(key)
            map[key] = Array.new
        end
        map[key].push(str)
    end
    map.each do |k,v|
        res.push(v)
    end
    return res
end
